geofileops/fileops.py

```diff
@@ -760,6 +760,15 @@ def rename_column(
             raise ValueError(f"rename_column not supported for {path}")
 
         # Rename column
+        if column_name.lower() == new_column_name.lower():
+            sql_stmt = (
+                f'ALTER TABLE "{layer}" '
+                f'RENAME COLUMN "{column_name}" TO "{column_name}_"'
```

Comment:
Probably safer to find a temporary column name in a loop while (case insensitively) checking that it doesn't exist yet in the list of current columns?